src/iguana/algorithms/clas12/FiducialFilter/Algorithm.h

```diff
@@ -0,0 +1,373 @@
+#pragma once
+
+#include "iguana/algorithms/Algorithm.h"
+#include "iguana/algorithms/TypeDefs.h"
+
+namespace iguana::clas12 {
+
+  /// @brief_algo Filter the `REC::Particle` bank by applying DC (drift chamber) fiducial cuts
+  ///
+  /// @begin_doc_algo{Filter}
+  /// @input_banks{REC::Particle, REC::Traj, RUN::config}
+  /// @output_banks{REC::Particle}
+  /// @end_doc
+  ///
+  /// @begin_doc_config
+  /// @config_param{pass | int | cook type to use for assigning fiducial cuts}
+  /// @end_doc
+  class FiducialFilter : public Algorithm
+  {
+
+      DEFINE_IGUANA_ALGORITHM(FiducialFilter, clas12::FiducialFilter)
+
+    public:
+
+      void Start(hipo::banklist& banks) override;
+      void Run(hipo::banklist& banks) const override;
+      void Stop() override;
+
+      
+      
+    private:
+      
+     /// Struct to store trajectory particle data
+     struct traj_row_data {
+        double x1 = -999;
+        double x2 = -999;
+        double x3 = -999;
+        double y1 = -999;
+        double y2 = -999;
+        double y3 = -999;
+        double z1 = -999;
+        double z2 = -999;
+        double z3 = -999;
+        int sector= 0;
+      };    
+    
+      /// @action_function{scalar filter} checks if the particle passes fiducial cuts
+      /// @param traj_row data struct of the particle in REC::Traj
+      /// @param torus toroidal magnetic field sign
+      /// @param pid pid of the particle
+      /// @returns `true` if passes fiducial cuts
+      bool Filter(FiducialFilter::traj_row_data const traj_row, int const torus, int const pid) const;
```

Comment:
This isn't an action function, since `traj_row` is not a simple scalar. Perhaps you need a vector action function here; I believe it would be the first (see #169). For this PR, I recommend:
- change the docstring so that it's not registering as an action function (remove `@action_function{scalar filter}`)
- open an issue saying `FiducialCuts` lacks an action function

Reply:
We need better documentation on what an action function is, exactly; added to https://github.com/JeffersonLab/iguana/issues/141